File: packages/jsii-pacmak/test/expected.java/java/org/jsii/tests/calculator/VirtualMethodPlayground.java
package org.jsii.tests.calculator;
@org.jsii.Jsii(module = org.jsii.tests.calculator.$Module.class, fqn = "jsii$jsii_calc$.VirtualMethodPlayground")
public class VirtualMethodPlayground extends org.jsii.JsiiObject {
    protected VirtualMethodPlayground(final org.jsii.JsiiObject.InitializationMode mode) {
        super(mode);
    }
    public VirtualMethodPlayground() {
        super(org.jsii.JsiiObject.InitializationMode.Jsii);
        org.jsii.JsiiEngine.getInstance().createNewObject(this);
    }
    public java.lang.Number serialSumAsync(final java.lang.Number count) {
        return this.jsiiAsyncCall("serialSumAsync", java.lang.Number.class, java.util.stream.Stream.of(count).toArray());
    }
    public java.lang.Number parallelSumAsync(final java.lang.Number count) {
        return this.jsiiAsyncCall("parallelSumAsync", java.lang.Number.class, java.util.stream.Stream.of(count).toArray());
    }
    public java.lang.Number sumSync(final java.lang.Number count) {
        return this.jsiiCall("sumSync", java.lang.Number.class, java.util.stream.Stream.of(count).toArray());
    }
    public java.lang.Number overrideMeAsync(final java.lang.Number index) {
        return this.jsiiAsyncCall("overrideMeAsync", java.lang.Number.class, java.util.stream.Stream.of(index).toArray());
    }
    public java.lang.Number overrideMeSync(final java.lang.Number index) {
        return this.jsiiCall("overrideMeSync", java.lang.Number.class, java.util.stream.Stream.of(index).toArray());
    }
}

File: packages/jsii-dotnet-runtime/test/AWS.Jsii.Tests.Calculator/VirtualMethodPlayground.cs
using AWS.Jsii.Runtime.Deputy;

namespace AWS.Jsii.Tests.Calculator
{
    [JsiiClass("jsii-calc", "jsii$jsii_calc$.VirtualMethodPlayground", "[]")]
    public class VirtualMethodPlayground : DeputyBase
    {
        public VirtualMethodPlayground(): base(new DeputyProps(new object[]{}))
        {
        }

        protected VirtualMethodPlayground(ByRefValue reference): base(reference)
        {
        }

        protected VirtualMethodPlayground(DeputyProps props): base(props)
        {
        }

        [JsiiMethod("serialSumAsync", "{\"primitive\":\"number\",\"promise\":true}", "[{\"name\":\"count\",\"type\":{\"primitive\":\"number\"}}]")]
        public virtual double SerialSumAsync(double count)
        {
            return InvokeMethod<double>(new object[]{count});
        }

        [JsiiMethod("parallelSumAsync", "{\"primitive\":\"number\",\"promise\":true}", "[{\"name\":\"count\",\"type\":{\"primitive\":\"number\"}}]")]
        public virtual double ParallelSumAsync(double count)
        {
            return InvokeMethod<double>(new object[]{count});
        }

        [JsiiMethod("sumSync", "{\"primitive\":\"number\"}", "[{\"name\":\"count\",\"type\":{\"primitive\":\"number\"}}]")]
        public virtual double SumSync(double count)
        {
            return InvokeMethod<double>(new object[]{count});
        }

        [JsiiMethod("overrideMeAsync", "{\"primitive\":\"number\",\"promise\":true}", "[{\"name\":\"index\",\"type\":{\"primitive\":\"number\"}}]")]
        public virtual double OverrideMeAsync(double index)
        {
            return InvokeMethod<double>(new object[]{index});
        }

        [JsiiMethod("overrideMeSync", "{\"primitive\":\"number\"}", "[{\"name\":\"index\",\"type\":{\"primitive\":\"number\"}}]")]
        public virtual double OverrideMeSync(double index)
        {
            return InvokeMethod<double>(new object[]{index});
        }
    }
}